### PR TITLE
Added first sketch of model parallelism and fixed multi-device intera…

### PR DESCRIPTION
**Pull request by dlibenzi**
…ction between PT/XLA and autograd.

### CODE REVIEW

File: torch_xla/csrc/tensor.cpp [hunk 1]
```diff
@@ -651,7 +651,12 @@ void XLATensor::zero_(XLATensor& input) {
 }
 
 void XLATensor::s_copy_(XLATensor& input, const XLATensor& src) {
-  input.SetIrValue(src.GetIrValue());
+  if (input.GetDevice() == src.GetDevice()) {
+    input.SetIrValue(src.GetIrValue());
+  } else {
+    XLA_ERROR() << "Cannot move from " << input.GetDevice() << " to "
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor.cpp]: This is not happening ATM. Will do in followup PR.

Comment (asuhan) [reply torch_xla/csrc/tensor.cpp]: What would we need this for? I see you went for `DataParallel` taking the training loop, so we shouldn't have to copy anything between devices.

File: torch_xla_py/data_parallel.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    return dqueue.queue.get()
+
+  def close(self):
+    self._done = True
+    for dqueue in itervalues(self._queues):
+      dqueue.queue.close()
+    self._loader_queue.close()
+
+  def _expand_sample_batch(self, data, target):
+    # TODO: Expand last sample,target to batch size
```
Comment (asuhan) [inline torch_xla_py/data_parallel.py]: I'm not sure how we'd do this, let the user specify the padding value? Could be something else than zero, at least in theory. We can find the correct one through graph analysis, longer term.

Comment (dlibenzi) [reply torch_xla_py/data_parallel.py]: Not critical, but many ways to do it.
1) Slice the last sample, broadcast in batchdim by (batch_size - last_batch_size), concat.
2) Slice the first (batch_size - last_batch_size), concat.
3) Keep batch-1, slice top (batch_size - last_batch_size), concat.

File: torch_xla/csrc/tensor.cpp (hunk 1 above)
Comment (dlibenzi) [reply torch_xla/csrc/tensor.cpp]: To make sure we can move data across devices with to().
Yes, today we do not use it, but it should be do crap like we'd do otherwise :)

Comment (asuhan) [reply torch_xla/csrc/tensor.cpp]: I see, definitely a good fix, just didn't see how it was related to this PR.
